packages/smoldot-test-utils/package.json

```diff
@@ -12,13 +12,15 @@
     "deep-clean": "rm -rf node_modules",
     "clean": "rm -rf dist/ tsconfig.tsbuildinfo",
     "build": "tsc -b --clean",
-    "test": "echo \"No tests for smoldot-test-utils\"",
+    "test": "exit 0; #This is a test support library",
     "lint": "eslint . --ext .js,.ts"
   },
   "dependencies": {
+    "asap": "^2.0.6",
```

Comment:
This is a very well-known battle tested library: It's written by a nodejs core contributor.  There's been no changes to it because its "done".  We can't use `setImmediate` because its non-standard and we shouldn't use `process.nextTick` because its a node specific API and we (rightly) get warnings about it.  It's the recommended (in MDN) way for tests to simulate yielding to the event loop to simulate asynchronous behaviour - this is exactly what we are doing here.